client/src/sagas/router.js

```diff
@@ -1,85 +1,81 @@
-import { push, LOCATION_CHANGE } from 'connected-react-router';
+import { LOCATION_CHANGE,push } from 'connected-react-router';
 import get from 'lodash/get';
 import {
-  all, select, put, takeLatest, delay, takeEvery,
-} from 'redux-saga/effects';
+  all, delay, put, select, takeEvery,
+takeLatest, } from 'redux-saga/effects';
 
 import * as actions from '../actions/type';
 import {
-  getPatientIdFromPatientPageRoute,
-  ROUTE_NAME_ROOT,
-  ROUTE_NAME_PATIENT,
-  PATIENT_SUBROUTE_SEARCH,
-  ROUTE_NAME_VARIANT,
+  Routes,
 } from '../helpers/route';
 
 function* navigateToVariantDetailsScreen(action) {
   try {
-    const { uid, tab } = action.payload;
-    let url = `${ROUTE_NAME_ROOT}${ROUTE_NAME_VARIANT}/${uid}`;
+    const { tab, uid } = action.payload;
+    let url = Routes.getPatientPath(uid);
     if (tab) { url += `/#${tab}`; }
 
     yield put(push(url));
   } catch (e) {
-    yield put({ type: actions.NAVIGATION_VARIANT_DETAILS_SCREEN_FAILED, message: e.message });
+    yield put({ message: e.message, type: actions.NAVIGATION_VARIANT_DETAILS_SCREEN_FAILED });
   }
 }
 
 function* navigateToPatientScreen(action) {
   try {
-    const { uid, tab, reload } = action.payload;
-    let url = `${ROUTE_NAME_ROOT}${ROUTE_NAME_PATIENT}/${uid}`;
-    if (tab) { url += `/#${tab}`; }
+    const { reload, tab, uid } = action.payload;
+    let url = Routes.getPatientPath(uid, tab);
+
     if (reload) {
       url += '?reload';
     }
 
-    yield put({ type: actions.PATIENT_SET_CURRENT_ACTIVE_KEY, payload: { activeKey: tab } });
+    yield put({ payload: { activeKey: tab }, type: actions.PATIENT_SET_CURRENT_ACTIVE_KEY });
     yield put(push(url));
   } catch (e) {
-    yield put({ type: actions.NAVIGATION_PATIENT_SCREEN_FAILED, message: e.message });
+    yield put({ message: e.message, type: actions.NAVIGATION_PATIENT_SCREEN_FAILED });
   }
 }
 
 function* navigateToPatientVariantScreen(action) {
   try {
     const { uid } = action.payload;
-    const url = `${ROUTE_NAME_ROOT}${ROUTE_NAME_PATIENT}/${uid}#variant`;
+    const url = Routes.getPatientPath(uid, 'variant');
 
     yield put(push(url));
   } catch (e) {
-    yield put({ type: actions.NAVIGATION_PATIENT_VARIANT_SCREEN_FAILED, message: e.message });
+    yield put({ message: e.message, type: actions.NAVIGATION_PATIENT_VARIANT_SCREEN_FAILED });
   }
 }
 
 function* navigateToSubmissionScreen() {
   try {
-    yield put(push(yield put(push('/submission'))));
+    yield put(push(yield put(push(Routes.Submission))));
```

Comment:
not your code but it's seems to me a bit far-fetched: yield => put => push => yield => push => push (same comment for all other occurrences in the page)